Game/Source/GDKTestGyms/AUptimeGameMode.cpp

```diff
@@ -447,37 +118,60 @@ void AUptimeGameMode::SpawnCrossServerActors(int32 CrossServerPointNum)
 		AUptimeCrossServerBeacon* Beacon = World->SpawnActor<AUptimeCrossServerBeacon>(CrossServerClass, Locations[i], FRotator::ZeroRotator, FActorSpawnParameters());
 		checkf(Beacon, TEXT("Beacon failed to spawn at %s"), *Locations[i].ToString());
 
-		SetCrossServerWorkerFlags(Beacon);
+		Beacon->SetCrossServerSize(CrossServerSize);
+		Beacon->SetCrossServerFrequency(CrossServerFrequency);
+		UE_LOG(LogUptimeGymGameMode, Log, TEXT("cross server size: %d, cross server frequency: %d"), CrossServerSize, CrossServerFrequency);
 	}
 }
 
 TArray<FVector> AUptimeGameMode::GenerateCrossServerLoaction()
 {
-	const float DistBetweenRows = ZoneHeight / SpawnRows;
-	const float DistBetweenCols = ZoneWidth / SpawnCols;
-	float StartingX = -(SpawnCols - 1) * ZoneWidth / 2 / SpawnCols;
-	float StartingY = -(SpawnRows - 1) * ZoneHeight / 2 / SpawnRows;
+	const float Height = GetZoneHeight();
+	const float Width = GetZoneWidth();
+	const float Rows = GetZoningRows();
+	const float Cols = GetZoningCols();
+
+	const float DistBetweenRows = Height / Rows;
```

Comment:
nit: semantics, but if we're dividing than that would suggest this is the WorldHeight, not the ZoneHeight

Reply:
but I see that ZoneHeight is actually the zone height - hmm
so we take the world height - divide by rows to get zone height, then divide by rows again to get DistBetweenRows?

Reply:
Actually this might be mistake. Previously we divided by SpawnRows/SpawnCols. These were set by worker flags. It may be the case these simply lined up with the number of ZoneRows / ZoneCols but I can't remember so will have a look.

Reply:
Yeah taking a look at what we have in TestGymBuildkite, SpawnCols and SpawnRows just lined up with how how many zone rows and zone columns there were in the test. This logic seems a little funky... I was just trying to reproduce what was there but... I don't think it makes sense for us to divide by the SpawnCols/SpawnRows (ZoningRows/ZoningCols) twice as you pointed out. I'll take a look at how to do this in a nicer way.

Reply:
Think I am overcomplicating thing... think we simply want one beacon spawned in the middle of each zone...